WordPress/src/main/java/org/wordpress/android/util/helpers/DynamicTextsProgressionHelper.kt

```diff
@@ -0,0 +1,100 @@
+package org.wordpress.android.util.helpers
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.animation.Animation
+import android.view.animation.Animation.AnimationListener
+import android.view.animation.AnimationUtils
+import android.widget.TextSwitcher
+import org.wordpress.android.R
+import org.wordpress.android.R.anim
+import java.lang.ref.WeakReference
+
+/**
+ * A simple helper to switch texts used to display progress e.g. of site being created.
+ * The helper is a runnable and once started (externally) will keep running until cancelled
+ * Each run changes the text to the next one in the sequence (stringIds) in a cyclic manner
+ *
+ * using a WeakReference for the text switcher will prevent future executions once the view is
+ * destroyed (or at least no longer referenced)
+ *
+ * @param textSwitcher - a week-ref to a text-switcher which displays the sequence
+ * @param stringIds - a list of string resource ids to sequence
+ * @param delay - how long to delay the next text-change (i.e. how long is each text displayed)
+ */
+class DynamicTextsProgressionHelper(
+    private val textSwitcher: WeakReference<TextSwitcher>,
+    private val stringIds: List<Int>,
+    val delay: Long
+) : Runnable {
+    val nText: Int = stringIds.size
+
+    // the count of how many times `nextText` was executed
+    var count = -1
```

Comment:
We might want to consider making all the fields private, wdyt?

Reply:
definitely. It's a remainder from when the class was `inner private`